best_practices.rst

```diff
@@ -0,0 +1,474 @@
+
+The Symfony binary is an executable command created in your machine when you
+`download Symfony`_. It provides multiple utilities, including the simplest way
+to create new Symfony applications:
+
+.. code-block:: terminal
+
+    $ symfony new my_project_name
+
+Under the hood, this Symfony binary command executes the needed `Composer`_
+command to create a new Symfony application based on the current stable version.
+
+Use the Symfony Skeleton to Create New Symfony-based Projects
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Symfony provides two "skeletons" (or "project templates") to
+:ref:`create new Symfony projects <creating-symfony-applications>`. Use the
+minimal "skeleton" to start small and get the best performance and grow your
+application later as needed.
+
+Use Composer and Symfony Flex to Manage Symfony Applications
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+`Composer`_ is the package manager used by modern PHP applications to manage
+their dependencies. :ref:`Symfony Flex <symfony-flex>` is a Composer plugin
+designed to automate some of the most common tasks performed in Symfony
+applications. Using Flex is optional but recommended because it improves your
+productivity significantly.
+
+Use the Directory Structure Proposed by Flex
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Unless your project follows a development practice that imposes a certain
+directory structure, follow the structure proposed by Symfony Flex. It's flat,
+self-explanatory and not coupled to Symfony:
+
+.. code-block:: text
+
+    your_project/
+    ├─ assets/
+    ├─ bin/
+    │  └─ console
+    ├─ config/
+    │  ├─ packages/
+    │  └─ services.yaml
+    └─ public/
+    │  ├─ build/
+    │  └─ index.php
+    ├─ src/
+    │  ├─ Kernel.php
+    │  ├─ Command/
+    │  ├─ Controller/
+    │  ├─ DataFixtures/
+    │  ├─ Entity/
+    │  ├─ EventSubscriber/
+    │  ├─ Form/
+    │  ├─ Migrations/
+    │  ├─ Repository/
+    │  ├─ Security/
+    │  └─ Twig/
+    ├─ templates/
+    ├─ tests/
+    ├─ translations/
+    ├─ var/
+    │  ├─ cache/
+    │  └─ log/
+    └─ vendor/
+
+Configuration
+-------------
+
+Use Environment Variables for Infrastructure Configuration
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+These are the options that change from one machine to another (e.g. from your
+development machine to the production server) but which don't change the
+application behavior.
+
+:ref:`Use env vars in your project <config-env-vars>` to define these options
+and create multiple ``.env`` files to :ref:`configure env vars per environment <config-dot-env>`.
+
+Use Parameters for Application Configuration
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+These are the options used to modify the application behavior, such as the sender
+of email notifications, or the enabled `feature toggles`_. Their value doesn't
+change per machine, so it's not worth it to define them as environment variables.
```

Comment:
so don't define them as ...

We should really emphasize the fact that env vars should be used only when strictly needed.